tuf/repository_lib.py

```diff
@@ -1608,14 +1619,36 @@ def generate_targets_metadata(targets_directory, target_files, version,
         if not os.path.exists(digest_target):
           logger.warning('Hard linking target file to ' + repr(digest_target))
           os.link(target_path, digest_target)
-  
-  # Generate the targets metadata object.
-  targets_metadata = tuf.formats.TargetsFile.make_metadata(version,
-                                                           expiration_date,
-                                                           filedict,
-                                                           delegations)
 
-  return targets_metadata
+  # Generate the targets metadata object.
+  # Use generalized build_dict_conforming_to_schema func to produce a dict that
+  # contains all the appropriate information for targets metadata,
+  # checking that the result conforms to the appropriate schema.
+  # TODO: Later, probably after the rewrite for TUF Issue #660, generalize
+  #       further, upward, by replacing generate_targets_metadata,
+  #       generate_root_metadata, etc. with one function that generates
+  #       metadata, possibly rolling that upwards into the calling function.
+  #       There are very few things that really need to be done differently.
+  if delegations is not None:
+    return tuf.formats.build_dict_conforming_to_schema(
+        tuf.formats.TARGETS_SCHEMA,
+        _type='Targets',   # TODO: Does this have to be capitalized? -.-
+        version=version,
+        expires=expiration_date,
+        targets=filedict,
+        delegations=delegations)
+  else:
+    return tuf.formats.build_dict_conforming_to_schema(
+        tuf.formats.TARGETS_SCHEMA,
+        _type='Targets',   # TODO: Does this have to be capitalized? -.-
+        version=version,
+        expires=expiration_date,
+        targets=filedict)
+  # TODO: As an alternative to the odd if/else above where we decide whether or
+  #       not to include the delegations argument based on whether or not it is
+  #       None, consider instead adding a check in
+  #       build_dict_conforming_to_schema that skips a keyword if that keyword
+  #       is optional in the schema and the value passed in is set to None....
```

Comment:
Here's another alternative: 

```python
...
extra = {}
if delegations is not None:
  extra = {"delegations": delegations}

return tuf.formats.build_dict_conforming_to_schema(
    tuf.formats.TARGETS_SCHEMA,
    _type='Targets',
    version=version,
    expires=expiration_date,
    targets=filedict,
    **extra)
```

Reply:
Hm.  That's interesting.  That might be a little too magic for this reference implementation.  I do like it, though.  Let's see if the need comes up more often.